src/bare-metal/alloc-example/src/main.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ANCHOR: Alloc
+#![no_main]
+#![no_std]
+
+extern crate alloc;
+extern crate panic_halt as _;
+
+use alloc::{string::ToString, vec::Vec};
+use buddy_system_allocator::LockedHeap;
+
+#[global_allocator]
+static HEAP_ALLOCATOR: LockedHeap<32> = LockedHeap::<32>::new();
+
+static mut HEAP: [u8; 65536] = [0; 65536];
+
+pub fn entry() {
+    // Safe because `HEAP` is only used here and `entry` is only called once.
+    unsafe {
+        // Give the allocator some memory to allocate.
+        HEAP_ALLOCATOR
```

Comment:
It looks like you could fit the whole thing on one line:

![image](https://user-images.githubusercontent.com/89623/222716222-a57c04bd-32f5-403c-bb8f-1d2996542102.png)

Reply:
True, but that's not how rustfmt formats it.

Reply:
Okay, fair enough! :smile: